File: create-icons.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

// Paths
const webPublic = path.join(__dirname, '..', 'trainjatri-web', 'public');
const appAssets = path.join(__dirname, 'assets', 'images');

// Brand colors
const LIGHT_BLUE = { r: 230, g: 244, b: 254 };
const FACEBOOK_BLUE = { r: 24, g: 119, b: 242 };
const WHITE = { r: 255, g: 255, b: 255 };

async function createIconWithBackground(logoPath, size, bgColor, padding) {
  const paddingPx = Math.floor(size * padding / 100);
  const logoSize = size - (2 * paddingPx);
  
  // Load and resize logo
  const logo = await sharp(logoPath)
    .resize(logoSize, logoSize, { fit: 'inside', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .toBuffer();
  
  // Get actual logo dimensions after resize
  const logoMetadata = await sharp(logo).metadata();
  
  // Calculate center position
  const x = Math.floor((size - logoMetadata.width) / 2);
  const y = Math.floor((size - logoMetadata.height) / 2);
  
  // Create background and composite logo
  return sharp({
    create: {
      width: size,
      height: size,
      channels: 3,
      background: bgColor
    }
  })
  .composite([{
    input: logo,
    left: x,
    top: y
  }]);
}

async function createTransparentIcon(logoPath, size, padding) {
  const paddingPx = Math.floor(size * padding / 100);
  const logoSize = size - (2 * paddingPx);
  
  // Load and resize logo
  const logo = await sharp(logoPath)
    .resize(logoSize, logoSize, { fit: 'inside', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .toBuffer();
  
  // Get actual logo dimensions after resize
  const logoMetadata = await sharp(logo).metadata();
  
  // Calculate center position
  const x = Math.floor((size - logoMetadata.width) / 2);
  const y = Math.floor((size - logoMetadata.height) / 2);
  
  // Create transparent background and composite logo
  return sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: { r: 0, g: 0, b: 0, alpha: 0 }
    }
  })
  .composite([{
    input: logo,
    left: x,
    top: y
  }]);
}

async function createMonochromeIcon(logoPath, size, padding) {
  const paddingPx = Math.floor(size * padding / 100);
  const logoSize = size - (2 * paddingPx);
  
  // Load, convert to grayscale, and resize logo
  const logo = await sharp(logoPath)
    .greyscale()
    .resize(logoSize, logoSize, { fit: 'inside', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .toBuffer();
  
  // Get actual logo dimensions
  const logoMetadata = await sharp(logo).metadata();
  
  // Calculate center position
  const x = Math.floor((size - logoMetadata.width) / 2);
  const y = Math.floor((size - logoMetadata.height) / 2);
  
  // Create white monochrome version
  const whiteLogo = await sharp(logo)
    .negate() // Invert to get white
    .toBuffer();
  
  // Create transparent background and composite
  return sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: { r: 0, g: 0, b: 0, alpha: 0 }
    }
  })
  .composite([{
    input: whiteLogo,
    left: x,
    top: y
  }]);
}

async function createIcons() {
  console.log('='.repeat(60));
  console.log('Creating TrainJatri App Icons');
  console.log('='.repeat(60));
  console.log('');
  
  const logoPath = path.join(webPublic, 'logo.png');
  
  // Check if logo exists
  if (!fs.existsSync(logoPath)) {
    console.error(`❌ Logo not found: ${logoPath}`);
    process.exit(1);
  }
  
  const logoMetadata = await sharp(logoPath).metadata();
  console.log(`✓ Using logo.png: ${logoMetadata.width}x${logoMetadata.height}`);
  console.log('');
  
  try {
    // 1. icon.png - Main app icon with white background
    console.log('1. Creating icon.png...');
    await (await createIconWithBackground(logoPath, 1024, WHITE, 15))
      .png()
      .toFile(path.join(appAssets, 'icon.png'));
    console.log('   ✓ icon.png saved (1024x1024, white background)');
    
    // 2. android-icon-foreground.png - Transparent for adaptive icon
    console.log('2. Creating android-icon-foreground.png...');
    await (await createTransparentIcon(logoPath, 1024, 20))
      .png()
      .toFile(path.join(appAssets, 'android-icon-foreground.png'));
    console.log('   ✓ android-icon-foreground.png saved (1024x1024, transparent)');
    
    // 3. android-icon-background.png - Solid color background
    console.log('3. Creating android-icon-background.png...');
    await sharp({
      create: {
        width: 1024,
        height: 1024,
        channels: 3,
        background: LIGHT_BLUE
      }
    })
    .png()
    .toFile(path.join(appAssets, 'android-icon-background.png'));
    console.log('   ✓ android-icon-background.png saved (1024x1024, light blue)');
    
    // 4. android-icon-monochrome.png - White monochrome version
    console.log('4. Creating android-icon-monochrome.png...');
    await (await createMonochromeIcon(logoPath, 1024, 20))
      .png()
      .toFile(path.join(appAssets, 'android-icon-monochrome.png'));
    console.log('   ✓ android-icon-monochrome.png saved (1024x1024, white monochrome)');
    
    // 5. splash-icon.png - Splash screen with transparency
    console.log('5. Creating splash-icon.png...');
    await (await createTransparentIcon(logoPath, 1024, 25))
      .png()
      .toFile(path.join(appAssets, 'splash-icon.png'));
    console.log('   ✓ splash-icon.png saved (1024x1024, transparent)');
    
    // 6. favicon.png - Small favicon with Facebook blue background
    console.log('6. Creating favicon.png...');
    await (await createIconWithBackground(logoPath, 48, FACEBOOK_BLUE, 10))
      .png()
      .toFile(path.join(appAssets, 'favicon.png'));
    console.log('   ✓ favicon.png saved (48x48, Facebook blue background)');
    
    console.log('');
    console.log('='.repeat(60));
    console.log('✨ All icons created successfully!');
    console.log('='.repeat(60));
    console.log('');
    console.log(`Icons saved to: ${appAssets}`);
    console.log('');
    console.log('Created files:');
    console.log('  - icon.png (1024x1024)');
    console.log('  - android-icon-foreground.png (1024x1024)');
    console.log('  - android-icon-background.png (1024x1024)');
    console.log('  - android-icon-monochrome.png (1024x1024)');
    console.log('  - splash-icon.png (1024x1024)');
    console.log('  - favicon.png (48x48)');
    console.log('');
    console.log('✓ Done! You can now delete the old React logo files:');
    console.log('  del assets\\images\\partial-react-logo.png');
    console.log('  del assets\\images\\react-logo.png');
    console.log('  del assets\\images\\react-logo@2x.png');
    console.log('  del assets\\images\\react-logo@3x.png');
    
  } catch (error) {
    console.error('❌ Error creating icons:', error);
    process.exit(1);
  }
}

// Run the icon creation
createIcons();

File: create_icons.py
from PIL import Image, ImageDraw, ImageOps
import os

# Paths
web_public = r"d:\train-jatri-app\trainjatri-web\public"
app_assets = r"d:\train-jatri-app\trainjatri-app\assets\images"

# Check existing logos
print("Checking existing logos from web app:")
for logo_file in ['logo.png', 'new_logo.png', 'final_1.png']:
    logo_path = os.path.join(web_public, logo_file)
    if os.path.exists(logo_path):
        img = Image.open(logo_path)
        print(f"{logo_file}: {img.size[0]}x{img.size[1]} pixels, mode: {img.mode}")
        img.close()

print("\n" + "="*60)
print("Creating app icons...")
print("="*60 + "\n")

# Load the main logo
logo_path = os.path.join(web_public, 'logo.png')
logo = Image.open(logo_path)
print(f"Using logo.png as base: {logo.size[0]}x{logo.size[1]}")

# Define brand colors
LIGHT_BLUE = "#E6F4FE"
FACEBOOK_BLUE = "#1877F2"
WHITE = "#FFFFFF"

def hex_to_rgb(hex_color):
    """Convert hex color to RGB tuple"""
    hex_color = hex_color.lstrip('#')
    return tuple(int(hex_color[i:i+2], 16) for i in (0, 2, 4))

def create_icon_with_background(logo_img, size, bg_color, padding_percent=15):
    """Create icon with logo centered on colored background"""
    # Create background
    icon = Image.new('RGB', size, hex_to_rgb(bg_color))
    
    # Calculate logo size with padding
    padding = int(size[0] * padding_percent / 100)
    logo_size = size[0] - (2 * padding)
    
    # Resize logo maintaining aspect ratio
    logo_resized = logo_img.copy()
    logo_resized.thumbnail((logo_size, logo_size), Image.Resampling.LANCZOS)
    
    # Calculate position to center logo
    x = (size[0] - logo_resized.width) // 2
    y = (size[1] - logo_resized.height) // 2
    
    # Paste logo (handle transparency if present)
    if logo_resized.mode == 'RGBA':
        icon.paste(logo_resized, (x, y), logo_resized)
    else:
        icon.paste(logo_resized, (x, y))
    
    return icon

def create_transparent_icon(logo_img, size, padding_percent=20):
    """Create icon with transparent background (for foreground layers)"""
    # Create transparent background
    icon = Image.new('RGBA', size, (0, 0, 0, 0))
    
    # Calculate logo size with padding
    padding = int(size[0] * padding_percent / 100)
    logo_size = size[0] - (2 * padding)
    
    # Resize logo maintaining aspect ratio
    logo_resized = logo_img.copy()
    if logo_resized.mode != 'RGBA':
        logo_resized = logo_resized.convert('RGBA')
    logo_resized.thumbnail((logo_size, logo_size), Image.Resampling.LANCZOS)
    
    # Calculate position to center logo
    x = (size[0] - logo_resized.width) // 2
    y = (size[1] - logo_resized.height) // 2
    
    # Paste logo
    icon.paste(logo_resized, (x, y), logo_resized)
    
    return icon

def create_monochrome_icon(logo_img, size, padding_percent=20):
    """Create white monochrome version for themed icons"""
    # Create transparent background
    icon = Image.new('RGBA', size, (0, 0, 0, 0))
    
    # Calculate logo size with padding
    padding = int(size[0] * padding_percent / 100)
    logo_size = size[0] - (2 * padding)
    
    # Convert logo to grayscale then to monochrome
    logo_gray = logo_img.copy().convert('L')
    logo_resized = logo_gray.resize((logo_size, logo_size), Image.Resampling.LANCZOS)
    
    # Create white version with alpha from grayscale
    logo_mono = Image.new('RGBA', logo_resized.size)
    pixels = logo_mono.load()
    gray_pixels = logo_resized.load()
    
    for y in range(logo_resized.height):
        for x in range(logo_resized.width):
            gray_val = gray_pixels[x, y]
            # Use gray value as alpha, make pixel white
            alpha = 255 if gray_val > 128 else 0
            pixels[x, y] = (255, 255, 255, alpha)
    
    # Calculate position to center
    x = (size[0] - logo_mono.width) // 2
    y = (size[1] - logo_mono.height) // 2
    
    # Paste on transparent background
    icon.paste(logo_mono, (x, y), logo_mono)
    
    return icon

# 1. Create icon.png (main app icon - 1024x1024, no transparency)
print("1. Creating icon.png...")
icon_main = create_icon_with_background(logo, (1024, 1024), WHITE, padding_percent=15)
icon_main.save(os.path.join(app_assets, 'icon.png'), 'PNG')
print("   ✓ icon.png saved (1024x1024, white background)")

# 2. Create android-icon-foreground.png (1024x1024, with transparency)
print("2. Creating android-icon-foreground.png...")
icon_foreground = create_transparent_icon(logo, (1024, 1024), padding_percent=20)
icon_foreground.save(os.path.join(app_assets, 'android-icon-foreground.png'), 'PNG')
print("   ✓ android-icon-foreground.png saved (1024x1024, transparent)")

# 3. Create android-icon-background.png (1024x1024, solid color)
print("3. Creating android-icon-background.png...")
bg_img = Image.new('RGB', (1024, 1024), hex_to_rgb(LIGHT_BLUE))
bg_img.save(os.path.join(app_assets, 'android-icon-background.png'), 'PNG')
print("   ✓ android-icon-background.png saved (1024x1024, light blue)")

# 4. Create android-icon-monochrome.png (1024x1024, white on transparent)
print("4. Creating android-icon-monochrome.png...")
icon_mono = create_monochrome_icon(logo, (1024, 1024), padding_percent=20)
icon_mono.save(os.path.join(app_assets, 'android-icon-monochrome.png'), 'PNG')
print("   ✓ android-icon-monochrome.png saved (1024x1024, white monochrome)")

# 5. Create splash-icon.png (1024x1024, with transparency for splash screen)
print("5. Creating splash-icon.png...")
splash_icon = create_transparent_icon(logo, (1024, 1024), padding_percent=25)
splash_icon.save(os.path.join(app_assets, 'splash-icon.png'), 'PNG')
print("   ✓ splash-icon.png saved (1024x1024, transparent)")

# 6. Create favicon.png (48x48 for web)
print("6. Creating favicon.png...")
favicon = create_icon_with_background(logo, (48, 48), FACEBOOK_BLUE, padding_percent=10)
favicon.save(os.path.join(app_assets, 'favicon.png'), 'PNG')
print("   ✓ favicon.png saved (48x48, Facebook blue background)")

print("\n" + "="*60)
print("✨ All icons created successfully!")
print("="*60)
print(f"\nIcons saved to: {app_assets}")
print("\nCreated files:")
print("  - icon.png (1024x1024)")
print("  - android-icon-foreground.png (1024x1024)")
print("  - android-icon-background.png (1024x1024)")
print("  - android-icon-monochrome.png (1024x1024)")
print("  - splash-icon.png (1024x1024)")
print("  - favicon.png (48x48)")

logo.close()
print("\n✓ Done!")
